src/upnp_xml.cc

```diff
@@ -238,6 +231,15 @@ std::unique_ptr<pugi::xml_document> UpnpXMLBuilder::renderDeviceDescription()
     for (auto&& [tag, field] : deviceProperties) {
         device.append_child(tag).append_child(pugi::node_pcdata).set_value(config->getOption(field).c_str());
     }
+    std::array<std::pair<const char*, const char*>, 4> deviceStringProperties { {
```

Comment:
I personally hate this.

Replacing all the entries with std::make_pair allows removal of extra {} and allows CTAD to work.